packages/braid-design-system/src/lib/components/Autosuggest/Autosuggest.tsx

```diff
@@ -346,14 +347,12 @@ export function highlightSuggestions(
   value: string,
   variant: HighlightOptions = 'matching',
 ): SuggestionMatch {
-  const matches = matchHighlights(suggestion, value);
+  const matches =
+    variant === 'matching'
+      ? matchHighlights(suggestion, value)
+      : reverseMatches(suggestion, matchHighlights(suggestion, value));
```

Comment:
I suggest extracting the matched highlights to a shared variable across both variants to guard against accidentally updating only one code path in the future

```suggestion
  const matchedHighlights = matchHighlights(suggestion, value);
  const matches =
    variant === 'matching'
      ? matchedHighlights
      : reverseMatches(suggestion, matchedHighlights);
```